src/components/CustomerDetails/CollectCustomerDetailsScreen.test.tsx

```diff
@@ -0,0 +1,273 @@
+import { Sentry } from "../../utils/errorTracking";
+import * as validateIdentification from "../../utils/validateIdentification";
+import { CampaignConfigsMap } from "../../context/campaignConfigsStore";
+import { CampaignConfigContextProvider } from "../../context/campaignConfig";
+import { AlertModalContextProvider, ERROR_MESSAGE } from "../../context/alert";
+import { CreateProvidersWrapper } from "../../test/helpers/providers";
+import { defaultFeatures, defaultProducts } from "../../test/helpers/defaults";
+import { CollectCustomerDetailsScreenContainer } from "./CollectCustomerDetailsScreen";
+import "../../common/i18n/i18nMock";
+import { IdentificationContextProvider } from "../../context/identification";
+
+jest.mock("../../utils/errorTracking");
+const mockCaptureException = jest.fn();
+(Sentry.captureException as jest.Mock).mockImplementation(mockCaptureException);
+
+const mockNavigate = { navigate: jest.fn() };
+const mockValidateAndCleanId = jest.spyOn(
+  validateIdentification,
+  "validateAndCleanId"
+);
+
+const identityDetailsInputTestId = "identity-details-input";
+const alternateIdInputTestId = "input-with-label-input";
+const checkButtonText = "Check";
+const passportCountryInputPlaceholderText = "Search country";
+const COUNTRY = "Afghanistan";
+
+jest.mock("react-navigation", () => ({
+  withNavigation: (Component: FunctionComponent) => (props: any) => (
+    <Component navigation={mockNavigate} {...props} />
+  ),
+  withNavigationFocus: (Component: FunctionComponent) => (props: any) => (
+    <Component navigation={mockNavigate} {...props} />
+  ),
+}));
+
+describe("CollectCustomerDetailsScreen", () => {
+  let allCampaignConfigs: CampaignConfigsMap;
+
+  beforeAll(() => {
+    allCampaignConfigs = {
+      campaignA: {
+        features: {
+          ...defaultFeatures,
+          id: {
+            label: "NRIC/FIN",
+            type: "STRING",
+            scannerType: "QR",
+            validation: "NRIC",
+          },
+          alternateIds: [
+            {
+              label: "Passport",
+              type: "STRING",
+              scannerType: "NONE",
+              validation: "PASSPORT",
+            },
+          ],
+          campaignName: "Some Campaign Name",
+        },
+        policies: [defaultProducts[0]],
+        c13n: {},
+      },
+    };
+  });
+
+  afterEach(() => {
+    cleanup();
+    jest.resetAllMocks();
+  });
+
+  describe("using nric", () => {
+    it("should navigate correctly with valid id input", async () => {
+      expect.assertions(5);
+      mockValidateAndCleanId.mockReturnValue("valid-id");
+
+      const { queryByText, queryByTestId } = render(
+        <CreateProvidersWrapper
+          providers={[
+            {
+              provider: CampaignConfigContextProvider,
+              props: { campaignConfig: allCampaignConfigs.campaignA },
+            },
+            { provider: IdentificationContextProvider },
+          ]}
```

Comment:
do you mean the sample campaignConfig? I was thinking to update `defaults.tsx` with some sample `campaignConfigs` instead since this config is not always mocked in all test cases